MetaMask/metamask-mobile
# refactor: consolidate accounts references to a single source of truth

## **Description**

1. What is the reason for the change?
The accounts team is working towards the integration of the new [accounts controller](https://github.com/MetaMask/core/tree/main/packages/accounts-controller). This controller will become the source of truth for all things accounts (name address, methods KeyringType etc). In order to make this integration easier we must first centralize the source of truth for accounts. 
3. What is the improvement/solution?
- Currently account information is being derived from the preferences controller. In this PR I removed direct use of `state.engine.backgroundState.PreferencesController` in favour of the respective redux selectors. I also migrated the Wallet screen to use the `useAccounts` hook which achieves the same result as before but without custom logic.

There should be no visible or functional changes to the app. 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/8482

## **Notes for code reviewers**
- 🔴  No use of `backgroundState.PreferencesController` `selectedAddress` and `identities`
- 🟢  Instead they should all be using the `selectIdentities` and the `selectSelectedAddress` selectors 

## **Manual testing steps**
There should be no visible changes after this PR

1. create/import a wallet
2. click the 3 dots on the the account selector 
3. change the account name
4. add an account
5. open the browser
6. navigate to a dapp: https://uniswap.org/
7. connect the dapp
8. ensure that the account names are the same as the ones you input previously

## **Screenshots/Recordings**



### **Before**



### **After**

Creation flow

https://github.com/MetaMask/metamask-mobile/assets/22918444/2afefbfa-55d5-446e-b00a-216b8daa2ec8

Import flow

https://github.com/MetaMask/metamask-mobile/assets/22918444/f9989e24-439a-4986-acd2-87c8229551a0


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
